### DISCUSSION TITLE
Updating Klipper on Kingroon KLP1 230

### DISCUSSION THREAD

**Discussion by Septain21** [mainsail-crew/mainsail#1985] [Help]
Good morning

I tried to update Klipper on my Kingroon kLP1 230 printer and I now get this error:

`MCU Protocol error

This is frequently caused by running an older version of the
firmware on the MCU(s). Fix by recompiling and flashing the
firmware.

Your Klipper version is: v0.12.0-286-g81de9a86
MCU(s) which should be updated:
mcu: Current version v0.11.0-122-ge6ef48cd-dirty-20230817_150801-mkspi
MKS_THR: Current version v0.11.0-122-ge6ef48cd-dirty-20231010_174111-mkspi
Up-to-date MCU(s):

Once the underlying issue is corrected, use the "RESTART"
command to reload the config and restart the host software.

mcu 'MKS_THR': Command format mismatch: query_adxl345 oid=%c rest_ticks=%u vs query_adxl345 oid=%c clock=%u rest_ticks=%u
`
I tried the following method by ssh accessing the printer:

`sudo service klipper stop
cd ~/klipper
git pull
make clean
make menuconfig
make
./scripts/flash-sdcard.sh /dev/ttyACM0 btt-octopus-f446-v1
[[[manually power-cycle the printer board here when instructed]]]
./scripts/flash-sdcard.sh -c /dev/ttyACM0 btt-octopus-f446-v1
sudo service klipper start
`

but here is the feedback I got:
`root@mkspi:~# cd ~/klipper
-bash: cd: /root/klipper: No such file or directory`

Have you seen a solution to offer me?

THANKS

**Comment by meteyou** [comment]
hey! I think your flash guide will generelly not fit, because it is for a BTT (BigTreeTech) board, but your printer use a MKS board. The MCU Protocol error also wrote, that the MKS_THR (the toolhead board and not the main board) has to be updated.

But, because this printer doesn't use a "generic klipper installation" (like MainsailOS or KIAUH installations are made), this is not very easy to debug for us.

So i would recommend you to ask the vendor support here or ask in a Kingroon close community for it.

**Reply by Septain21** [reply]
Thank you for your support

**Comment by oboronshik** [comment]
Ты залогинился как "root", а надо как "mks"...

You logged in as "root", but you should log in as "mks"...

**Reply by meteyou** [reply]
pls write english only here...
